spacegraph/spacegraph_codebase/model.py

```diff
@@ -54,13 +53,24 @@ def forward(self, ng_list, do_full_eval = True):
             neg_embeds: the N negative sampled center points feature embedding
                     shape (batch_size, num_neg_sample, embed_dim)
         '''
-        # random sample each context points in NeighborGraph()
-        self.sample_context_pts(ng_list)
-
         if do_full_eval == False:
             # random sample each context points in NeighborGraph()
             self.sample_neg_pts(ng_list)
 
+        center_pred_embed = self.predict(ng_list)
```

Comment:
In this file, I added a `predict` function for each model, which only embeds the point itself and not the positive and negative points. The `predict` function is then used in the `forward` functions. This looks like a lot of changes but it's mainly encapsulation of the center_pred_embed part